dataclients/kubernetes/clusterstate_test.go

```diff
@@ -0,0 +1,60 @@
+package kubernetes
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/zalando/skipper/dataclients/kubernetes/definitions"
+)
+
+var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func randStringRunes(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letterRunes[rand.Intn(len(letterRunes))]
+	}
+	return string(b)
+}
+
+func benchmarkCachedEndpoints(b *testing.B, n int) {
+	endpoints := make(map[definitions.ResourceID]*endpoint)
+	for i := 0; i < n; i++ {
+		name := randStringRunes(10)
+		if i == 0 {
+			name = "postgres"
+		}
```

Comment:
This is also redundant, target endpoint name could be `fmt.Sprintf("foo-%d", n / 2)` or even `"foo-0"`. Lets make it as simple as possible.